src/components/Auth/AuthForm.tsx

```diff
@@ -0,0 +1,154 @@
+import classNames from 'classnames';
+import React, { useEffect, useState } from 'react';
+import { createUser, getUserByEmail } from '../../api/users';
+import { User } from '../../types/User';
+
+export type Props = {
+  onLogin: (user: User) => void,
+};
+
+export const AuthForm: React.FC<Props> = ({ onLogin }) => {
+  const [email, setEmail] = useState('');
+  const [name, setName] = useState('');
+  const [needToRegister, setNeedToRegister] = useState(false);
+  const [loading, setLoading] = useState(false);
+  const [errorMessage, setErrorMessage] = useState('');
+
+  const saveUser = (user: User) => {
+    localStorage.setItem('user', JSON.stringify(user));
```

Comment:
It is better to create a seperateService to work with Storage and use it in components to inverse our dependencies.  It will be much easier to replace localStorage with SessionStorage for example